File: frontend/src/types/llm.ts
export type LLMProviderName = "openai" | "anthropic" | "litellm";

export interface ProviderModels {
  provider: LLMProviderName;
  models: string[];
  configured: boolean;
}

export interface LLMDefault {
  provider: LLMProviderName;
  model: string;
}

export interface ApiKeyStatus {
  provider: LLMProviderName;
  configured: boolean;
  source: "database" | "environment" | "none";
  api_base: string | null;
}

export interface ApiKeyUpdate {
  api_key: string;
  api_base?: string;
}

File: app/schemas/llm.py
from typing import Literal

from pydantic import BaseModel

LLMProviderName = Literal["openai", "anthropic", "litellm"]


class ProviderModels(BaseModel):
    provider: LLMProviderName
    models: list[str]
    configured: bool


class LLMDefault(BaseModel):
    provider: LLMProviderName
    model: str


class ApiKeyStatus(BaseModel):
    provider: LLMProviderName
    configured: bool
    source: Literal["database", "environment", "none"]
    api_base: str | None = None


class ApiKeyUpdate(BaseModel):
    api_key: str
    api_base: str | None = None
